pkg/sidecar_mounter/sidecar_mounter.go

```diff
@@ -239,6 +247,17 @@ func collectMetrics(ctx context.Context, port, tempDir string) {
 
 	// Create a unix domain socket and listen for incoming connections.
 	socketPath := filepath.Join(tempDir, metrics.SocketName)
+
+	// Cleanup any stale metrics path to handle container restart scenario.
+	if _, err := os.Stat(socketPath); err == nil {
+		klog.V(4).Infof("found stale socket path for metrics collector %s, attempting delete of the path", socketPath)
```

Comment:
please create a playbook entry for metrics collector where the metrics may be malformed across node reboots